src/community/features-templating/features-templating-web/src/test/java/org/geoserver/featurestemplating/web/TemplateRuleConfigurationPanelTest.java

```diff
@@ -169,4 +169,26 @@ public Component buildComponent(final String id) {
         tester.assertErrorMessages(
                 "The Template extension and the chosen output format are incompatible.");
     }
+
+    @Test
+    public void testLayerRuleProfileCQL() {
+        LayerInfo layerInfo = getCatalog().getLayerByName(getLayerId(MockData.BUILDINGS));
+        Model<LayerInfo> layerModel = new Model<>(layerInfo);
+        tester.startPage(
+                new FormTestPage(
+                        new ComponentBuilder() {
+                            private static final long serialVersionUID = -5907648151984337786L;
+
+                            @Override
+                            public Component buildComponent(final String id) {
+                                return new TemplateRulesTabPanel(id, layerModel);
+                            }
+                        }));
+        FormTester form = tester.newFormTester("form:panel:ruleConfiguration:theForm");
+        form.select("templateIdentifier", 0);
+        form.select("outputFormats", 0);
+        form.setValue("profileFilter", "header('Contet-Profile')='http://myProfile'");
+        form.submit("save");
+        tester.assertNoErrorMessage();
```

Comment:
Maybe also check the update actually happened in the catalog?

Reply:
the test class will test only the panel not the general page. That's why there is no additional check to the catalog, in the scope of the test the rule is added to the listview only.